packages/flutter_test/lib/src/event_simulation.dart

```diff
@@ -902,8 +902,13 @@ Future<bool> simulateKeyDownEvent(
   String? platform,
   PhysicalKeyboardKey? physicalKey,
   String? character,
-}) {
-  return KeyEventSimulator.simulateKeyDownEvent(key, platform: platform, physicalKey: physicalKey, character: character);
+}) async {
+  final bool handled = await KeyEventSimulator.simulateKeyDownEvent(key, platform: platform, physicalKey: physicalKey, character: character);
```

Comment:
I think we should just verify the selectors are handled correctly instead of trying to verify the key presses are handled correctly.

Reply:
`TestWidgetsFlutterBinding.testTextInput` is not nullable. So if binding is `TestWidgetsFlutterBinding` the text input should be present. 

The purpose of this code is to turn unhandled key events into selectors, if text input is currently active, same way `FlutterTextInputPlugin` does it on macOS. There are no keyboard shortcuts defined for macOS anymore, everything is driven by selectors, so Flutter Tester must be able to simulate the environment, otherwise most text editing tests just fail on macOS.

That said, I don't know if `event_simulation` is the right place to do this.

Reply:
I think this process is reasonable. The entire test binding is to emulate a real device, and this new logic reflects our new model for the macOS system, which translates unhandled key events into selectors.